From 8368c860e2e7949dae175abb5bb87cd2e4bf8206 Mon Sep 17 00:00:00 2001
From: "annulen@yandex.ru"
 <annulen@yandex.ru@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Wed, 18 Oct 2017 14:57:13 +0000
Subject: [PATCH] REGRESSION(r217771): nullTerminatedWCharToString loses last
 character of input string https://bugs.webkit.org/show_bug.cgi?id=178444

Reviewed by Per Arne Vollan.

* wtf/text/win/WCharStringExtras.h:
(WTF::nullTerminatedWCharToString):


git-svn-id: http://svn.webkit.org/repository/webkit/trunk@223606 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 Source/WTF/wtf/text/win/WCharStringExtras.h |  2 +-
 1 files changed, 1 insertions(+), 1 deletion(-)

diff --git a/Source/WTF/wtf/text/win/WCharStringExtras.h b/Source/WTF/wtf/text/win/WCharStringExtras.h
index 9d004e51595..d41f46294d3 100644
--- a/Source/WTF/wtf/text/win/WCharStringExtras.h
+++ b/Source/WTF/wtf/text/win/WCharStringExtras.h
@@ -60,7 +60,7 @@ inline String wcharToString(const wchar_t* characters, unsigned length)
 
 inline String nullTerminatedWCharToString(const wchar_t* characters)
 {
-    return wcharToString(characters, wcslen(characters) - 1);
+    return wcharToString(characters, wcslen(characters));
 }
 
 } // namespace WTF
-- 
2.16.1

